src/components/navigation/DragonsList.js

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchDragons } from '../../redux/dragons/dragonsSlice';
+import Dragons from '../Dragons';
+
+function DragonsList() {
+  const { dragons, isLoading } = useSelector((state) => state.dragons);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchDragons());
+  }, [dispatch]);
+
```

Comment:
- [ ] Please make sure that the data is fetched only once and it should not be added to the store on every re-render (i.e. when changing views / using navigation).  Currently when ever I switch to a different page the previous page data is re-rendered and the whole page is updating to the initial state. See the video comparison for more explanation:

Your website:

https://github.com/iamsjunaid/space-travelers-hub/assets/44798044/7c58771b-e3b1-48db-a63b-fff253afe113

Expected:

https://github.com/iamsjunaid/space-travelers-hub/assets/44798044/cc3c137a-d545-4c6a-841a-306ccce6e1b4